### PR TITLE
cmake: Bump max LLVM version to 19

### PR DESCRIPTION
**Pull request by kraj**
##### Checklist

- [ ] Language changes are updated in `man/adoc/bpftrace.adoc`
- [ ] User-visible and non-trivial changes updated in `CHANGELOG.md`
- [ ] The new behaviour is covered by tests

### CODE REVIEW

**Review by viktormalik — COMMENTED** [review]
Do I get it right that this is finally an LLVM release where all we have to do is bump the version? Incredible :open_mouth:

Anyways, I'd prefer if we waited for Nix to add LLVM 19 so that we can run all the tests. Since it should be released tomorrow, it should hopefully not take long before it gets to Nixpkgs.

Also a changelog entry would be nice.

**Comment by kraj** [comment]
> Do I get it right that this is finally an LLVM release where all we have to do is bump the version? Incredible 😮
> 

yes :)

> Anyways, I'd prefer if we waited for Nix to add LLVM 19 so that we can run all the tests. Since it should be released tomorrow, it should hopefully not take long before it gets to Nixpkgs.

yocto has clang-19 staged already and waiting for final release
https://github.com/kraj/meta-clang/pull/982
and has been testing this patch

> 
> Also a changelog entry would be nice.

OK

**Comment by danobi** [comment]
LGTM,  but like Viktor said, would be good to wait until we have it in CI until we merge this. Cuz if we merge and we cut a release, might ship a buggy release.

**Comment by kraj** [comment]
> LGTM, but like Viktor said, would be good to wait until we have it in CI until we merge this. Cuz if we merge and we cut a release, might ship a buggy release.

No issues, sounds reasonable.

**Review by danobi — APPROVED** [review]

**Review by viktormalik — APPROVED** [review]

**Comment by danobi** [comment]
Looks like `ubuntu:devel` is already on llvm 19. So our distro build fails in CI.

I'm thinking about merging this but gating llvm 19 behind CMAKE_BUILD_TYPE=Debug. Idea is that we don't want llvm 19 built by packagers yet but also want CI working until nix releases llvm 19.

Or we could move CI to track nix unstable. Kinda hesitant to do this b/c nix is kinda a black box for many folks and I'm not sure we want to be debugging weird nix things.

**Comment by ajor** [comment]
> I'm thinking about merging this but gating llvm 19 behind CMAKE_BUILD_TYPE=Debug. Idea is that we don't want llvm 19 built by packagers yet but also want CI working until nix releases llvm 19.

Sounds like a good idea to me. Also lets people experiment with LLVM 19 before we declare it as officially supported.

**Review by viktormalik — COMMENTED** [review]
Still missing the CI job in `.github/workflows/ci.yml` 🙂

**Comment by danobi** [comment]
Needs this patch applied to flake.nix: https://pastes.dxuuu.xyz/z5unr7

File: .github/workflows/ci.yml
Comment (viktormalik) [inline .github/workflows/ci.yml:106]: For the sake of limiting the amount of CI jobs, we tend to test all LLVM versions in the `Debug` and all the additional configs (`Release`, Clang, AOT, memleak) with just the latest LLVM.

Could you please do that? In other words, this PR should just add one new job `LLVM 19 Debug` and change LLVM 18->19 in all of the other jobs.

**Review by viktormalik — CHANGES_REQUESTED** [review]

**Comment by viktormalik** [comment]
> Needs this patch applied to flake.nix: https://pastes.dxuuu.xyz/z5unr7

Still needs this for the LLVM 19 jobs to pass.

**Review by viktormalik — APPROVED** [review]
Thanks!

Finally LLVM 19 support :rocket:

**Pull request merged**
